api/src/main/scala/JobUtils.scala

```diff
@@ -0,0 +1,140 @@
+package org.wikiwatershed.mmw.geoprocessing
+
+import java.util.concurrent.atomic.LongAdder
+import collection.mutable
+import collection.concurrent.TrieMap
+import spray.json._
+import DefaultJsonProtocol._
+import geotrellis.proj4.{CRS, ConusAlbers, LatLng, WebMercator}
+import geotrellis.raster._
+import geotrellis.raster.rasterize._
+import geotrellis.raster.rasterize.Rasterizer.Options
+import geotrellis.raster.mapalgebra.focal.Kernel
+import geotrellis.vector._
+import geotrellis.vector.io._
+import geotrellis.spark._
+import geotrellis.spark.io._
+import geotrellis.spark.io.s3._
+
+trait JobUtils {
+  val baseLayerReader = S3CollectionLayerReader("datahub-catalogs-us-east-1", "")
+
+  /**
+    * For an InputJson object, return a histogram of raster grouped count results.
+    *
+    * @param   crs    The key (e.g. "input.rasterCRS")
+    * @param   input  The InputJson
+    * @return         A histogram of results in JSON format
+    */
+  def getRasterGroupedCount(input: InputJson) = {
+    val rasterCRS = getCRS(input.rasterCRS)
+    val polygonCRS = getCRS(input.polygonCRS)
+    val rasterLayerIds = input.rasters.map { str => LayerId(str, input.zoom) }
+    val polygons = input.polygons.map { str =>
+      parseGeometry(str, polygonCRS, rasterCRS).buffer(0).asMultiPolygon.get
+    }
+    val aoi = polygons.unionGeometries.asMultiPolygon.get
+    val rasterLayers = rasterLayerIds.map { layer => fetchLayer(layer, aoi) }
+    val count =
+      rasterGroupedCount(rasterLayers, aoi)
+        .mapValues(_.sum().toInt)
+        .map { case (k, v) => k.toString -> v}
+        .toMap
```

Comment:
I think this `toMap` is necessary to turn the mutable map into an immutable map for Spray serialization.